client/dashboard/components/date-range-picker/date-range-content.tsx

```diff
@@ -0,0 +1,218 @@
+import { DateInputs } from './date-inputs';
+import { PresetsListbox } from './presets-listbox';
+import { computePresetRange, getActivePresetId, PresetId } from './utils';
+
+type DateRangeContentProps = {
+	isSmall: boolean;
+	fromDraft?: Date;
+	toDraft?: Date;
+	fromStr: string;
+	toStr: string;
+	setFromDraft: ( date?: Date ) => void;
+	setToDraft: ( date?: Date ) => void;
+	setFromStr: ( string: string ) => void;
+	setToStr: ( string: string ) => void;
+	timezoneString?: string;
+	gmtOffset?: number;
+	onChange: ( next: { start: Date; end: Date } ) => void;
+	onClose?: () => void;
+	compositeActiveId: string | null;
+	setCompositeActiveId: ( id: string | null ) => void;
+	siteToday: Date;
+	siteTodayStr: string;
+	mobileLabelId: string;
+	desktopLabelId: string;
+};
+
+const today = new Date();
+( today as Date ).setHours( 0, 0, 0, 0 );
+
+export function DateRangeContent( props: DateRangeContentProps ) {
+	const {
+		isSmall,
+		fromDraft,
+		toDraft,
+		fromStr,
+		toStr,
+		setFromDraft,
+		setToDraft,
+		setFromStr,
+		setToStr,
+		timezoneString,
+		gmtOffset,
+		onChange,
+		onClose,
+		compositeActiveId,
+		setCompositeActiveId,
+		siteToday,
+		siteTodayStr,
+		mobileLabelId,
+		desktopLabelId,
+	} = props;
+
+	const clear = () => {
+		setFromDraft( undefined );
+		setToDraft( undefined );
+		setFromStr( '' );
+		setToStr( '' );
+	};
+
+	const apply = () => {
+		if ( ! fromDraft || ! toDraft ) {
+			return;
+		}
+		const [ startPoint, endPoint ] =
+			fromDraft <= toDraft ? [ fromDraft, toDraft ] : [ toDraft, fromDraft ];
+		onChange( { start: startPoint, end: endPoint } );
+		onClose?.();
+	};
+
+	const setPreset = ( id: PresetId ) => {
+		const range = computePresetRange( id, siteToday );
+		if ( ! range ) {
+			return;
+		}
+		setFromDraft( range.from );
+		setToDraft( range.to );
+		setFromStr( formatYmd( range.from, timezoneString, gmtOffset ) );
+		setToStr( formatYmd( range.to, timezoneString, gmtOffset ) );
+		onChange( { start: range.from, end: range.to } );
+		onClose?.();
+	};
+
+	const activePresetId = getActivePresetId( fromDraft, toDraft, siteToday );
+
+	const defaultMonth = isSmall ? today : new Date( today.getFullYear(), today.getMonth() - 1, 1 );
+	const endMonth = new Date( today.getFullYear(), today.getMonth(), 1 );
+
+	// Use TZDate for calendar selection when an IANA time zone is available
+	const selected =
+		timezoneString && ( fromDraft || toDraft )
+			? {
+					from: fromDraft ? new TZDate( +fromDraft, timezoneString ) : undefined,
+					to: toDraft ? new TZDate( +toDraft, timezoneString ) : undefined,
+			  }
+			: { from: fromDraft, to: toDraft };
+
+	return (
+		<div style={ { padding: 12 } }>
+			<Text as="div" weight={ 600 } align="center" size="smallTitle" style={ { marginBottom: 8 } }>
+				{ __( 'Date Range' ) }
+			</Text>
+
+			{ isSmall ? (
+				<>
+					<VStack
+						as="div"
+						spacing={ 1 }
+						className="daterange-presets"
+						style={ { marginBottom: 16 } }
```

Comment:
Instead of using marginBottom, would it be better to add another VStack and set the spacing 🤔

Reply:
Also, can we move VStack to `PresetsListbox` itself? It seems the component needs it internally 🤔